frontend/src/components/PlanHome/SearchBar.tsx

```diff
@@ -0,0 +1,74 @@
+import { useState } from 'react'
+import { type Contact } from '../../utils/types'
+import styles from './SearchBar.module.css'
+
+interface Props {
+  handleSelectChangeFn: (data: Contact) => void
+  data: Contact[]
+  alreadySelectedData: Contact[]
+}
+
+export const SearchBar = ({
+  handleSelectChangeFn,
+  data,
+  alreadySelectedData
+}: Props): JSX.Element => {
+  const [query, setQuery] = useState<string>('')
+  const [searchResult, setSearchResult] = useState<Contact[]>(data)
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputString: string = event.target.value
+    setQuery(inputString)
+
+    const filteredSearchByQuery = data.filter(
+      (contact: Contact) => contact.email.toLowerCase().startsWith(inputString)
+    )
+
+    const filteredSearchByAlreadySelected = filteredSearchByQuery.filter(
+      (contact: Contact) => !alreadySelectedData.includes(contact)
+    )
+
+    console.log(filteredSearchByAlreadySelected)
+
+    setSearchResult(filteredSearchByAlreadySelected)
+  }
+
+  const handleClearResults = () => {
+    setQuery('')
+  }
+
+  const handleSelect = (contact: Contact) => {
+    if (alreadySelectedData.includes(contact)) {
+      return
+    }
+    handleSelectChangeFn(contact)
+  }
+  return (
+    <div className={styles.searchBarContainer}>
+      <input
+        type="text"
+        placeholder="Search by contact email"
+        className={styles.searchBar}
+        onChange={handleQueryChange}
+        value={query}/>
+      {query.length > 0 &&
+        <button className={styles.clear} onClick={handleClearResults}>clear</button>
+      }
+      <div className={`${styles.searchResultContainer} ${styles.scrollable}`}>
```

Comment:
consider using a utility like classnames: https://www.npmjs.com/package/classnames

this is a kinda ugly way of applying styles to a div by using string interpolation